cli/JSXV4.md

```diff
@@ -22,9 +22,10 @@ To build certain dependencies in V3 compatibility mode, whatever the version use
 
 ```json
 "jsx": {
- "version": 4,
- "v3-dependencies": ["rescript-react-native", "rescript-react-navigation"]
-}
+  "version": 3,
+  "v3-dependencies": ["rescript-react-native", "rescript-react-navigation"]
```

Comment:
This flag specifies on what dependencies v3 is activated. It's not useful if the default version is 3.

You might be thinking of a different examples, where everything builds on v3?

Reply:
Actually it does also add `-open ReactV3`.

There seem so the 2 use cases:
- Take an existing project and build it with latest rescript-react. Question is: why would you do that if the old rescript-react works and this is the only change. But it can be a stepping stone.
- Migrate a project towards V4 without converting all the dependencies to V3 first.

Reply:
I think we have to look into it because I have something strange. Here is the repro example.
* The root project with rc-3 version of compiler and rescript-react. The jsx version config in bsconfig is 3.
* The root project has dependency which is not compatible to v4, e.g. rescript-relay@1.0.2

The rescript-relay is supposed to be built as v3, but it wouldn't. It throws the error:
```
FAILED: src/RescriptRelay.cmj

  We've found a bug for you!
  /Users/woonki/Github/works/gl/sources/sinsunhi-frontend-mirror/node_modules/rescript-relay/src/RescriptRelay.res:666:9-75

  664 ┆   React.createElement(
  665 ┆     provider,
  666 ┆     {"value": Some({"environment": environment}), "children": children
      ┆ },
  667 ┆   )
  668 ┆ }

  This has type:
    {"children": 'a, "value": option<{"environment": Environment.t}>}
  Somewhere wanted: React.Context.props<option<contextShape>>
```
If I add the `v3-dependencies: ["rescript-relay"]`, the error is gone.

Maybe, the jsx config in the root project is not propagated to the dependency which has the old config `"reason": {"react": 3}` until it is added to `v3-dependencies`, I guess.

Reply:
We had a same discussion before https://github.com/rescript-lang/rescript-react/pull/49#issuecomment-1253530706

Reply:
Updated. How does it look?

Reply:
Thanks! It looks much better.
Still wondering how much the second use case is useful though. Because the root project could not use the component from the v3 dependencies which doesn’t have a props type.
But it is stating more informative example and explanation now!